File: RxOperators/app/src/main/java/me/gurpreetsk/rxoperators/ui/MainActivity.java
package me.gurpreetsk.rxoperators.ui;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import butterknife.ButterKnife;
import butterknife.OnClick;
import me.gurpreetsk.rxoperators.R;
import me.gurpreetsk.rxoperators.ui.combination.CombinationOperatorsActivity;
import me.gurpreetsk.rxoperators.ui.filtering.FilteringOperatorsActivity;
import me.gurpreetsk.rxoperators.ui.mathematical.MathematicalOperatorsActivity;
import me.gurpreetsk.rxoperators.ui.transformation.TransformationOperatorsActivity;

public class MainActivity extends AppCompatActivity {

  private static final String TAG = MainActivity.class.getSimpleName();

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    ButterKnife.bind(this);
  }

  @OnClick(R.id.button_transformation)
  public void openTransformationActivity() {
    startActivity(new Intent(MainActivity.this, TransformationOperatorsActivity.class));
  }

  @OnClick(R.id.button_combinational)
  public void openCombinationalActivity() {
    startActivity(new Intent(MainActivity.this, CombinationOperatorsActivity.class));
  }

  @OnClick(R.id.button_filtering)
  public void openFilteringActivity() {
    startActivity(new Intent(MainActivity.this, FilteringOperatorsActivity.class));
  }

  @OnClick(R.id.button_mathematical)
  public void openMathematicalActivity() {
    startActivity(new Intent(MainActivity.this, MathematicalOperatorsActivity.class));
  }

}

File: RxOperators/app/src/main/kotlin/me/gurpreetsk/rxoperators/ui/MainActivity.kt
package me.gurpreetsk.rxoperators.ui

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import butterknife.ButterKnife
import kotlinx.android.synthetic.main.activity_main.*
import me.gurpreetsk.rxoperators.R
import me.gurpreetsk.rxoperators.ui.combination.CombinationOperatorsActivity
import me.gurpreetsk.rxoperators.ui.filtering.FilteringOperatorsActivity
import me.gurpreetsk.rxoperators.ui.mathematical.MathematicalOperatorsActivity
import me.gurpreetsk.rxoperators.ui.transformation.TransformationOperatorsActivity

class MainActivity : AppCompatActivity() {
  private val TAG = MainActivity::class.java.simpleName

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)
    ButterKnife.bind(this)

    button_transformation.setOnClickListener({
      startActivity(Intent(this@MainActivity, TransformationOperatorsActivity::class.java))
    })
    button_combinational.setOnClickListener({
      startActivity(Intent(this@MainActivity, CombinationOperatorsActivity::class.java))
    })
    button_filtering.setOnClickListener({
      startActivity(Intent(this@MainActivity, FilteringOperatorsActivity::class.java))
    })
    button_mathematical.setOnClickListener({
      startActivity(Intent(this@MainActivity, MathematicalOperatorsActivity::class.java))
    })
  }
}
